### PR TITLE
Improve Testing

### PR DESCRIPTION
**Pull request by MarJMue**
- Test for all currently maintained python versions
- Add a basic test to check if jupyter decorators obviously break

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -10,7 +10,53 @@ on:
     branches: [ master ]
 
 jobs:
-  build:
+  test_python_37:
```
Comment (domna) [inline .github/workflows/pytest.yml]: Could be simplified to one job using a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs).

**Review by domna — COMMENTED** [review]
The nbmake should just test that ipynb runs w/o errors, right? However, I think it actually should throw an error in the master branch, because the decorator fails for the matplotlib backend (which is default if not otherwise specified), because it is injected in FigureWidget. So I don't know if some assertions need to be added to the nb to check if it runs w/o errors.

**Comment by domna** [comment]
Ah sorry, I didn't see that you cherry-picked the fix :)

**Comment by MarJMue** [comment]
Yes pytest with nbmake throws an error, if any cell breaks. Visual checks of the interactive output are a lot more complicated, but it provides at least a simple check if the code executes. 

The main goal of the PR was to examine if the bug you fixed (thanks btw :+1: ) was somehow version dependent.

**Pull request merged**